mqttclients/dotnet/MQTTnet.Client.Extensions/TelemetryConsumer.cs

```diff
@@ -27,10 +27,16 @@ public async Task StartAsync()
                 var segments = topic.Split('/');
                 var msg = new TelemetryMessage<T>()
                 {
+                    MessageId = m.PacketIdentifier,
                     ClientIdFromTopic = segments[1],
-                    Payload = _serializer.FromBytes<T>(m.ApplicationMessage.Payload)
+                    Payload = _serializer.FromBytes<T>(m.ApplicationMessage.PayloadSegment.ToArray())
                 };
-                OnTelemetryReceived?.Invoke(msg);
+                bool accept = await OnTelemetryReceived?.Invoke(msg)!;
```

Comment:
Are we adding 'ConfigureAwait(false)` to async calls to avoid tying a call to a specific thread?

Reply:
I dont think we need it, this is useful when the library is going to be used in UI applications, for these samples I dont think we need to go so deep into async/await internals